vNext/UnityGLTFv2.cs

```diff
@@ -0,0 +1,238 @@
+    /// </summary>
+    /// <param name="gltfObject">Object which contains information to parse</param>
+    /// <param name="textureId">Texture to load from glTF object.</param>        
+    /// <returns>The created Unity object</returns>
+    public virtual Task<Texture2D> ImportTextureAsync(
+        IGLTFObject gltfObject,
+        int textureId,
+        CancellationToken cancellationToken = CancellationToken.None
+        );
+}
+
+// UnityNode.cs
+public virtual partial class Importer
+{
+    private virtual Task<GameObject> ConstructNode();
+}
+
+// UnityTexture.cs
+public virtual partial class Importer
+{
+    private virtual Task<Texture2D> ConstructTexture();
+}
+
+public class ImporterConfig
+{
+    public ImporterConfig();
+    public ImporterConfig(List<IUnityExtensionFactory> registry, GLTFImportOptions importOptions);
+}
+
+/// <summary>
+/// Rename of ILoader. Now returns tasks instead of IEnumerator.
+/// Handles the reading in of data from a path
+/// </summary>
+public class IDataLoader
+{
+    Task<Stream> LoadStreamAsync(string uri, CancellationToken ct = CancellationToken.None);
+}
+
+public class GLTFImportOptions
+{
+    /// <summary> Scheduler of tasks. Can be replaced with custom app implementation so app can handle background threads </summary>
+    System.Threading.Tasks.TaskScheduler TaskScheduler { get; set; }
+
+    /// <summary>Override for the shader to use on created materials </summary>
+    string CustomShaderName { get; set; }
+
+    /// <summary> Adds colliders to primitive objects when created </summary>
+    ColliderType Collider { get; set; }
+}
+
+public partial class Exporter
+{
+    /// <param name="dataWriter">Interface for handling the streams of data to write out</param>
+    /// <param name="progress">Progress of export</param>
+    public Exporter(
+        IDataWriter dataWriter,
+        ExporterConfig exportConfig = new ExporterConfig()
+        );
+
+    /// <summary>
+    /// Exports a Unity object to a glTF file
+    /// </summary>
+    /// <param name="unityObject">The object to export</param>
+    /// <param name="exportConfig">Configuration of extension settings and export options</param>
+    /// <returns>Strongly typed version of exported object</returns>
+    public Task<IGLTFObject> ExportAsync(
+        GameObject unityObject,
+        IProgress<int> progress = null,
+        CancellationToken ct = CancellationToken.None
+        );
+}
+
+/// <summary>
+/// Writes data for an export operation
+/// </summary>
+public class IDataWriter
+{
+    Task<bool> WriteStreamAsync(string uri, Stream stream, CancellationToken ct = CancellationToken.None);
+}
+
+public class ExporterConfig
+{
+    public ExporterConfig();
+    public ExporterConfig(List<IUnityExtensionFactory> registry, GLTFExportOptions exportOptions);
+}
+
+public class GLTFExportOptions
+{
+    /// <summary>Whether to write the object out as a GLB</summary>
+    bool ShouldWriteGLB { get; set ; }

```

Comment:
Does it not make more sense for this to just be a parameter of the actual Export* method(s)? Is it not possible to have a single exporter instance that exports both gltf and glb?